File: Samples/Layout/HelloWorld/C#/HelloWorld6.cs
using System;
using System.Globalization;
using System.IO;

using BitMiracle.Docotic.Pdf.Layout;

namespace BitMiracle.Docotic.Pdf.Samples
{
    class HelloWorld6
    {
        public static void CreatePdf()
        {
            PdfDocumentBuilder.Create().Generate("hello.pdf", BuildDocument);
        }

        static void BuildDocument(Document doc)
        {
            doc.Typography(t =>
            {
                t.Document = doc.TextStyleWithFont(SystemFont.Family("Calibri"));
            });

            var imageFile = new FileInfo(@"..\Sample Data\red-flowers-at-butterfly-world.jpg");
            var image = doc.Image(imageFile);

            doc.Pages(pages => {
                BuildPages(pages);

                pages.Content().Column(c =>
                {
                    BuildTextContent(c.Item());
                    BuildImageContent(c.Item(), image);
                    BuildListContent(c.Item());
                });
            });
        }

        static void BuildListContent(LayoutContainer content)
        {
            var dayNames = DateTimeFormatInfo.InvariantInfo.DayNames;
            var dayNamesSpain = DateTimeFormatInfo.GetInstance(new CultureInfo("es-ES")).DayNames;

            content.Column(column =>
            {
                for (int i = 0; i < dayNames.Length; i++)
                {
                    column.Item().Row(row =>
                    {
                        row.Spacing(5);
                        row.AutoItem().Text($"{i + 1}.");
                        row.RelativeItem().Text(t => {
                            t.Line(dayNames[i]);
                            t.Line($"In Spain they call it {dayNamesSpain[i]}");
                        });
                    });
                }
            });
        }

        static void BuildImageContent(LayoutContainer content, Image image)
        {
            content.AlignCenter()
                .PaddingVertical(20)
                .Image(image);
        }

        static void BuildPages(PageLayout pages)
        {
            pages.Size(PdfPaperSize.A6);
            pages.Margin(10);

            BuildPagesHeader(pages.Header());
            BuildPagesFooter(pages.Footer());
        }

        static void BuildTextContent(LayoutContainer content)
        {
            var colorAccented = new PdfRgbColor(56, 194, 10);
            var styleAccented = TextStyle.Parent.FontColor(colorAccented);

            content.Text(t =>
            {
                t.Span("Hello, ").Style(styleAccented);
                t.Span("world!").Style(styleAccented.Underline());
            });
        }

        static void BuildPagesHeader(LayoutContainer header)
        {
            header.TextStyle(TextStyle.Parent.FontSize(8))
                .AlignRight()
                .Text(t =>
                {
                    t.Line($"Created by: {Environment.UserName}");
                    t.Line($"Date: {DateTime.Now}");
                });
        }

        static void BuildPagesFooter(LayoutContainer footer)
        {
            footer.Height(20)
                .Background(new PdfGrayColor(95))
                .AlignCenter()
                .Text(t => t.CurrentPageNumber());
        }
    }
}

File: Samples/Layout/HelloWorld/VB.NET/HelloWorld6.vb
Imports System.Globalization
Imports System.IO
Imports BitMiracle.Docotic.Pdf.Layout

Namespace BitMiracle.Docotic.Pdf.Samples
    Class HelloWorld6
        Public Shared Sub CreatePdf()
            PdfDocumentBuilder.Create().Generate("hello.pdf", AddressOf BuildDocument)
        End Sub

        Private Shared Sub BuildDocument(doc As Document)
            doc.Typography(Sub(t) t.Document = doc.TextStyleWithFont(SystemFont.Family("Calibri")))

            Dim imageFile = New FileInfo("..\Sample Data\red-flowers-at-butterfly-world.jpg")
            Dim image = doc.Image(imageFile)

            doc.Pages(
                Sub(pages)
                    BuildPages(pages)

                    pages.Content().Column(
                        Sub(c)
                            BuildTextContent(c.Item())
                            BuildImageContent(c.Item(), image)
                            BuildListContent(c.Item())
                        End Sub)
                End Sub)
        End Sub

        Private Shared Sub BuildListContent(content As LayoutContainer)
            Dim dayNames = DateTimeFormatInfo.InvariantInfo.DayNames
            Dim dayNamesSpain = DateTimeFormatInfo.GetInstance(New CultureInfo("es-ES")).DayNames

            content.Column(
                Sub(column)
                    For i As Integer = 0 To dayNames.Length - 1
                        Dim rowIndex = i
                        column.Item().Row(
                            Sub(row)
                                row.Spacing(5)
                                row.AutoItem().Text($"{rowIndex + 1}.")
                                row.RelativeItem().Text(
                                    Sub(t)
                                        t.Line(dayNames(rowIndex))
                                        t.Line($"In Spain they call it {dayNamesSpain(rowIndex)}")
                                    End Sub)
                            End Sub)
                    Next
                End Sub)
        End Sub

        Private Shared Sub BuildImageContent(content As LayoutContainer, image As Image)
            content.AlignCenter().
                PaddingVertical(20).
                Image(image)
        End Sub

        Private Shared Sub BuildPages(pages As PageLayout)
            pages.Size(PdfPaperSize.A6)
            pages.Margin(10)
            BuildPagesHeader(pages.Header())
            BuildPagesFooter(pages.Footer())
        End Sub

        Private Shared Sub BuildTextContent(content As LayoutContainer)
            Dim colorAccented = New PdfRgbColor(56, 194, 10)
            Dim styleAccented = TextStyle.Parent.FontColor(colorAccented)
            content.Text(
                Sub(t)
                    t.Span("Hello, ").Style(styleAccented)
                    t.Span("world!").Style(styleAccented.Underline())
                End Sub)
        End Sub

        Private Shared Sub BuildPagesHeader(header As LayoutContainer)
            header.TextStyle(TextStyle.Parent.FontSize(8)).
                AlignRight().Text(
                    Sub(t)
                        t.Line($"Created by: {Environment.UserName}")
                        t.Line($"Date: {DateTime.Now}")
                    End Sub)
        End Sub

        Private Shared Sub BuildPagesFooter(footer As LayoutContainer)
            footer.Height(20).
                Background(New PdfGrayColor(95)).
                AlignCenter().
                Text(Sub(t) t.CurrentPageNumber())
        End Sub
    End Class
End Namespace
